spec/requests/custom_buttons_spec.rb

```diff
@@ -147,6 +183,44 @@
       expect(cb.reload.visibility[:roles]).to eq(['_ALL_'])
       expect(cb.reload.resource_action.ae_attributes).to include("attribute" => "is present")
     end
+
+    it 'updates the existing resource action' do
+      cb.resource_action = ResourceAction.create!(:dialog_id => 4)
+      api_basic_authorize collection_action_identifier(:custom_buttons, :edit)
+
+      request = {
+        'action'    => 'edit',
+        'resources' => [
+          {'id' => cb.id.to_s, 'resource' => {'name' => 'updated 1', 'resource_action' => {'dialog_id' => 2}}},
+
+        ]
+      }
+
+      expect(cb.resource_action.dialog_id).to eq(4)
+
+      post(api_custom_buttons_url, :params => request)
+      cb.reload
+
+      expect(ResourceAction.first.dialog_id).to eq(2)
```

Comment:
same here

Reply:
You mean line 199, right? This one should be fine.

Reply:
Both?  This is so strange, why would you be able to update that attribute in this way?

Reply:
Because I have a ticket in which someone expressly requested this functionality?

Reply:
It's not really that strange, being able to update or create a button and specify the dialog seems perfectly reasonable to me really.

Reply:
I guess the part I'm having trouble with it that we're updating attributes of a `ResourceAction` through the `/api/custom_buttons/:id` URL rather than `/api/custom_buttons/:id/resource_action` (doesn't exist) or `/api/resource_actions/:id` (only supports GET).  So it's hard to validate the attributes that are allowed to be changed throught the API.

In theory you could update `resource_action["resource_id"]` to the ID of a different CustomButton that already has a ResourceAction and totally break the `has_one` relation.  (We should probably have a model validation for that)

Reply:
Okay ... but can that be done in later work, please?